src/main/java/edu/cornell/kfs/sys/cache/CuEhCacheCacheManager.java

```diff
@@ -0,0 +1,191 @@
+package edu.cornell.kfs.sys.cache;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+
+import net.sf.ehcache.CacheManager;
+import net.sf.ehcache.Ehcache;
+import net.sf.ehcache.event.NotificationScope;
+import net.sf.ehcache.event.RegisteredEventListeners;
+
+public class CuEhCacheCacheManager extends EhCacheCacheManager implements RedisEventListener, DisposableBean {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private CacheManager ehcacheManager;
+    private RedisEventListenerLazyInitProxy redisEventListenerProxy;
+    private EhcacheEventListenerForUpdatingRedis ehcacheEventListener;
+    private Set<String> cacheNames;
+    private Set<String> cachesIgnoringRedisEvents;
+    private Set<String> cachesToClearOnRedisConnectionChange;
+
+    private Set<String> redisAwareCacheNames;
+
+    @Override
+    public void afterPropertiesSet() {
+        Objects.requireNonNull(ehcacheManager, "Internal Ehcache Manager cannot be null");
+        Objects.requireNonNull(redisEventListenerProxy, "redisEventListenerProxy cannot be null");
+        Objects.requireNonNull(ehcacheEventListener, "ehcacheEventListener cannot be null");
+        Objects.requireNonNull(cacheNames, "cacheNames cannot be null");
+        Objects.requireNonNull(cachesIgnoringRedisEvents, "cachesIgnoringRedisEvents cannot be null");
+        Objects.requireNonNull(cachesToClearOnRedisConnectionChange,
+                "cachesToClearOnRedisConnectionChange cannot be null");
+        
+        if (!cacheNames.containsAll(cachesIgnoringRedisEvents)) {
+            throw new IllegalStateException("cachesIgnoringRedisEvents references caches not found in cacheNames");
```

Comment:
This might be an improvement for later, but would it be helpful have debug  log statements specifying the cachesIgnoringRedisEvents or cachesToClearOnRedisConnectionChange that aren't in cacheNames

Reply:
Yes, having that addition as an enhancement makes sense.